keps/sig-api-machinery/1965-kube-apiserver-identity/README.md

```diff
@@ -154,64 +197,147 @@ Alpha should provide basic functionality covered with tests described above.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Stale Lease objects will be garbage collected.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, see [apiserver_identity_test.go](https://github.com/kubernetes/kubernetes/blob/24238425492227fdbb55c687fd4e94c8b58c1ee3/test/integration/controlplane/apiserver_identity_test.go).
 
 ### Rollout, Upgrade and Rollback Planning
 
-_This section must be completed when targeting beta graduation to a release._
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Existing workloads should not be impacteded by this feature, unless they were
+looking for Lease objects in the `kube-apiserver-lease` namespace.
+
+###### What specific metrics should inform a rollback?
+
+Recently added [healthcheck metrics for apiserver](https://github.com/kubernetes/kubernetes/pull/112741), which includes
+the health of the post start hook can be used to inform rollback, specifically `kubernetes_healthcheck{poststarthook/start-kube-apiserver-identity-lease-controller}`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Manual testing for upgrade/rollback will be done prior to Beta. Steps taken for manual tests will be updated here.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
 
 ### Monitoring Requirements
 
-_This section must be completed when targeting beta graduation to a release._
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The existence of the `kube-apiserver-lease` namespace and Lease objects in the namespace
+will determine if the feature is working. Operators can check for clients that are accessing
+the Lease object to see if workloads or other controllers are relying on this feature.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason:
+- [X] API .status
+  - Condition name:
+  - Other field:
+- [X] Other (treat as last resort)
+  - Details: audit logs for clients that are reading the Lease objects
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+A rough SLO here is that kube-apiserver updates leases at the same frequency as kubelet node heart beats,
+since the same mechanism is being used.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [X] Metrics
+  - Metric name: kubernetes_healthcheck
+  - [Optional] Aggregation method: name="poststarthook/start-kube-apiserver-identity-lease-controller"
+  - Components exposing the metric: kube-apiserver
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Yes, heart beat latency could be useful.
```

Comment:
LastUpdateTime of my own lease object.
But that has a potential cardinality problem, given the lease is changed on every kube-apiserver restart...

We could consider something like "oldest lease", but given we leave old leases for a longer time (1h), that's not particularly useful...

So I'm not entirely sure how we could construct such a metric here...

Reply:
Yeah, my understanding for this section was "metrics that would be nice but we wouldn't added due to cardinality issues, etc"

Reply:
Updated so it's clear we're not actually going to add this metric